src/lib/factory.js

```diff
@@ -38,30 +38,30 @@ class PluginFactory extends EventEmitter2 {
     return co.wrap(this._connect).call(this, options)
   }
   * _connect (options) {
-    if (this.adminPlugin) return yield this.adminPlugin.connect(options)
-
-    // create the central admin instance
-    this.adminPlugin = new Plugin({
-      username: this.adminUsername,
-      password: this.adminPassword,
-      account: this.adminAccount,
-      prefix: this.configPrefix
-    })
-    this.adminPlugin.removeAllListeners('_rpc:notification')
-    this.adminPlugin.on('_rpc:notification', (notif) =>
-      co.wrap(this._routeNotification).call(this, notif))
+    if (!this.adminPlugin) {
+      // create the central admin instance
+      this.adminPlugin = new Plugin({
+        username: this.adminUsername,
+        password: this.adminPassword,
+        account: this.adminAccount,
+        prefix: this.configPrefix
+      })
+      this.adminPlugin.removeAllListeners('_rpc:notification')
+      this.adminPlugin.on('_rpc:notification', (notif) =>
+        co.wrap(this._routeNotification).call(this, notif))
+
+      this.adminPlugin.on('connect', () => {
```

Comment:
This should be wrapped in a promise and yielded at some point, so that factory.connect won't resolve until after `this._subscribeAccounts()` is called in this callback.